src/utils/Codec.ts

```diff
@@ -32,11 +38,51 @@ export class Codec {
         ["Dispute", DisputeEthersType]
     ]);
 
-    public static encode(struct: StructType): string {
-        const structName = struct.constructor.name;
-        const ethersType = this.structToEthersType.get(structName);
+    public static encode(
+        struct: StructType,
+        explicitType?: StructTypeName
+    ): string {
+        let ethersType: string | undefined;
+
+        // If explicit type is provided, use it directly
+        if (explicitType) {
+            ethersType = this.structToEthersType.get(explicitType);
+        } else {
+            // Try constructor name first
+            const structName = struct.constructor.name;
+            ethersType = this.structToEthersType.get(structName);
+
+            // Simple fallback for plain objects - try common types
+            if (!ethersType && structName === "Object") {
+                // Try JoinChannel first (most common)
+                if (
+                    "channelId" in struct &&
+                    "participant" in struct &&
+                    "balance" in struct
+                ) {
+                    ethersType = JoinChannelEthersType;
+                }
+                // Try Block
+                else if ("transaction" in struct && "stateHash" in struct) {
+                    ethersType = BlockEthersType;
+                }
+                // Try Transaction
+                else if ("header" in struct && "body" in struct) {
+                    ethersType = TransactionEthersType;
+                }
+                // Try Dispute
+                else if ("disputeIndex" in struct && "channelId" in struct) {
+                    ethersType = DisputeEthersType;
```

Comment:
Fallback doesn't make sense if you already predefine what 'can be encoded' and also 'explicitType' doesn't make sense if you define what can be encoded -> remove the fallback (let it throw if you try encoding an invalid type) and split the encode into 2 functions that are overloaded:
encode(struct: StructType) -> which has all the info to deduct the correct encoding
encode(struct: any, explicitType: StructTypeName) -> useful if the object is not an instance of the predefined class -> try and encode it to the explicitType - it will fail if it can't encode